components/mdns/mdns.c

```diff
@@ -6329,8 +6341,14 @@ esp_err_t mdns_service_remove_all(void)
 /*
  * MDNS QUERY
  * */
-
 void mdns_query_results_free(mdns_result_t *results)
```

Comment:
A question: should we consider this API be called in other task? This API might also called in the notifer of mdns search once callback, this should be a mdns context, but the user might now be call this API directly in this callback, so a more safe way is that we double check the task handle, if not a mdns task, we might need to take a lock. What's your idear @david-cermak ?